lib/modules/manager/pep621/processors/pdm.ts

```diff
@@ -17,7 +17,7 @@ import { PdmLockfileSchema, type PyProject } from '../schema';
 import { depTypes, parseDependencyGroupRecord } from '../utils';
 import type { PyProjectProcessor } from './types';
 
-const pdmUpdateCMD = 'pdm update --no-sync';
+const pdmUpdateCMD = 'pdm update --no-sync --update-eager';
```

Comment:
I am not sure the exact version that supports this feature, but I am sure it is not a new feature. It might be supoprted even several years ago. In the [changelog](https://github.com/pdm-project/pdm/blob/main/CHANGELOG.md#release-v2112-2024-01-02) there is a bug fix for this feature in version v2.11.2 on 2024-01-02.

Reply:
January 2024 is still pretty recent. If users might be using a version of pdm which doesn't support this flag, and which would error because of it, then we need to allow for that

Reply:
I just tried pdm version 2.0.0 which is release almost two years ago (Jul 15, 2022) and it already has `--update-eager` option for `pdm update`.

Reply:
I can't work out exactly how old it is but it does seem pretty old. Also, we don't seem to automatically detect pdm constraints anyway and would only use an old version if the user explicitly configured it in constraints.pdm